js/webtorrent/components/torrentFileList.js

```diff
@@ -40,18 +40,21 @@ class TorrentFileList extends React.Component {
   }
 
   renderFileLink (file, isDownload) {
-    const { torrent, torrentID } = this.props
+    const { torrentId, torrent, serverUrl } = this.props
     const ix = torrent.files.indexOf(file)
     if (isDownload) {
-      if (torrent.serverURL) {
-        const httpURL = torrent.serverURL + '/' + ix
+      if (serverUrl) {
+        const httpURL = serverUrl + '/' + ix
         return <a href={httpURL} download={file.name}>⇩</a>
       } else {
         return <div /> // No download links until the server is ready
       }
     } else {
-      const magnetURL = torrentID + '&ix=' + ix
-      return <a href={magnetURL}>{file.name}</a>
+      const suffix = /^https?:/.test(torrentId)
+        ? '#ix=' + ix
+        : '&ix=' + ix
```

Comment:
Fix race condition where media is briefly added to <iframe>  …
If there is no file object yet, but there is a serverUrl, then an
iframe is added to the page, instead of a "Loading..." message